daemon/graphdriver/btrfs/btrfs.go

```diff
@@ -311,6 +302,36 @@ func subvolDelete(dirpath, name string, quotaEnabled bool) error {
 	return nil
 }
 
+func subvolSetPropRO(path string, isReadOnly bool) error {
+	dir, err := openDir(path)
+	if err != nil {
+		return err
+	}
+	defer closeDir(dir)
+
+	var oldflags, newflags C.__u64
+	_, _, errno := unix.Syscall(unix.SYS_IOCTL, getDirFd(dir), C.BTRFS_IOC_SUBVOL_GETFLAGS,
+		uintptr(unsafe.Pointer(&oldflags)))
+	if errno != 0 {
+		return fmt.Errorf("Failed to get btrfs subvolume flags for %s: %v", path, errno.Error())
+	}
+
+	if isReadOnly {
+		newflags = oldflags | C.BTRFS_SUBVOL_RDONLY
+	} else {
+		newflags = oldflags &^ C.BTRFS_SUBVOL_RDONLY
+	}
+
+	if newflags != oldflags {
```

Comment:
style nitpick: better to have

```go
if newflags == oldflags {
	return nil
}
```

while at it, maybe use Go naming convention (i.e. `oldFlags`, `newFlags`).

Reply:
naming convention is OK, but sorry, I don't get why `newflags == oldflags` is better. It's not error to stop function immediately. `newflags != oldflags` clearly indicate optional part and we still have to `return nil` at the end.